src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Machine.class) {
+            return List.of();
+        }
+
+        try {
+            try {
+                return List.of(type.getDeclaredConstructor().newInstance());
```

Comment:
Using reflection to instantiate objects can lead to performance issues and security risks. Consider using a factory pattern or dependency injection if possible.